maple-core/src/generic_node/vdom.rs

```diff
@@ -0,0 +1,180 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fmt;
+use std::rc::Rc;
+
+use crate::generic_node::{EventListener, GenericNode};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum Node {
```

Comment:
Continuing from my previous review, how about making `template!` render to a `Node` and use `Node` to generate real DOM. There isn't any VDOM overhead as it is only used on creation.